src/main/groovy/spock/genesis/generators/composites/PermutationGenerator.groovy

```diff
@@ -0,0 +1,97 @@
+package spock.genesis.generators.composites
+
+import groovy.transform.CompileStatic
+import spock.genesis.generators.Generator
+import spock.genesis.generators.UnmodifiableIterator
+
+@CompileStatic
+class PermutationGenerator<E> extends TupleGenerator<E> {
+
+    static final int MAX_PERMUTATIONS = 10000
+    private final int maxDepth
+
+    PermutationGenerator(List<Generator<E>> generators, int maxDepth) {
+        super(generators)
+        this.maxDepth = maxDepth
+    }
+
+    PermutationGenerator(List<Generator<E>> generators) {
+        super(generators)
+        this.maxDepth = pickMaxDepth(generators)
+    }
```

Comment:
Change

```groovy
 PermutationGenerator(List<Generator<E>> generators) {
         super(generators)
         this.maxDepth = pickMaxDepth(generators)
 }
```

to

```groovy
 PermutationGenerator(List<Generator<E>> generators) {
         this(generators, pickMaxDepth(generators))
 }
```

Then you will be reusing `PermutationGenerator(List<Generator<E>> generators, int maxDepth)`, which at the moment was not covered by any test. From now on it will :)